core/shared/src/main/scala/zio/FiberFailure.scala

```diff
@@ -27,10 +27,20 @@ import zio.stacktracer.TracingImplicits.disableAutoTrace
  * better integrate with Scala exception handling.
  */
 final case class FiberFailure(cause: Cause[Any]) extends Throwable(null, null, true, false) {
+
+  private var javaStackTrace: Array[StackTraceElement] = Thread.currentThread().getStackTrace
+
+  def this(cause: Cause[Any], javaStackTrace: Array[StackTraceElement]) = {
+    this(cause)
+    this.javaStackTrace = javaStackTrace
+  }
+
   override def getMessage: String = cause.unified.headOption.fold("<unknown>")(_.message)
 
-  override def getStackTrace(): Array[StackTraceElement] =
-    cause.unified.headOption.fold[Chunk[StackTraceElement]](Chunk.empty)(_.trace).toArray
+  override def getStackTrace(): Array[StackTraceElement] = {
+    val zioStackTrace = cause.unified.headOption.fold[Chunk[StackTraceElement]](Chunk.empty)(_.trace).toArray
+    zioStackTrace ++ javaStackTrace
+  }
```

Comment:
Isn't it simpler to just change L29 so that the Throwable contains the stack trace?

```scala
final case class FiberFailure(cause: Cause[Any]) extends Throwable(null, null, true, true)
```

And then:

```scala
  override def getStackTrace(): Array[StackTraceElement] = {
    val zioStackTrace = cause.unified.headOption.fold[Chunk[StackTraceElement]](Chunk.empty)(_.trace).toArray
    zioStackTrace ++ super.getStackTrace()
  }
```

Reply:
Note that this should be optimized. Currently it allocates 3 arrays but I think it can be rewritten to allocate just 2 (maybe even just 1)

Reply:
Done!